src/cmd/services/m3query/config/config.go

```diff
@@ -128,6 +128,10 @@ type Configuration struct {
 
 	// Cache configurations.
 	Cache CacheConfiguration `yaml:"cache"`
+
+	// DropNaNs drops NaNs before returning query results.
+	// If you want to simulate Promtheus exactly, set this to true.
+	DropNaNs bool `yaml:"dropNaNs"`
```

Comment:
Don't have this as a top level config setting

Reply:
What do you propose? Maybe have like a `QueryOptions` struct?

Reply:
`Result` would make more sense imo; also should default to true

Reply:
Maybe rename it to `returnNansInQueryResult` or something

Reply:
+1 on `ResultOptions` (or `Result`)